pkg/identity/types.go

```diff
@@ -26,6 +25,31 @@ func (si ServiceIdentity) String() string {
 	return string(si)
 }
 
+// GetSDSCSecretName returns a string key used as the name of Certificate in all SDS structs.
+func (si ServiceIdentity) GetSDSCSecretName() string {
```

Comment:
I don't think we should be leaking SDS implementation details in this generic package meant for identity. Instead, I would prefer a helper in the SDS package to convert the ServiceIdentity type to an SDS secret name.

Reply:
@draychev Is this a temporary change? Will a subsequent change remove SDS related code from this package?

Reply:
Made https://github.com/openservicemesh/osm/issues/3182 and added a comment.
I would like to be able to use `serviceIdentity.GetSDSSecretName()` to make the follow up PRs smaller.
After we are done transitioning to `ServiceIdentity` - I'll remove these from the `identity` package.